core/src/main/scala/org/apache/spark/serializer/SerializationDebugger.scala

```diff
@@ -110,11 +112,11 @@ private[spark] object SerializationDebugger extends Logging {
             val elem = s"externalizable object (class ${e.getClass.getName}, $e)"
             visitExternalizable(e, elem :: stack)
 
-          case s: Object with java.io.Serializable =>
+          case s: Object with java.io.Serializable if Utils.isTesting =>
             val str = try {
               String.valueOf(s)
```

Comment:
Can we just call `s.toString` ? We had already checked that`o != null` before

Reply:
Follow suggestion